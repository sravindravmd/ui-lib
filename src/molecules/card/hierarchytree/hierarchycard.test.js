import React from 'react'
import { render, fireEvent, cleanup } from '@testing-library/react'
import HierarchyTreeCard from './index'

afterEach(cleanup)

describe('<HierarchyTreeCard/>', () => {
  it('renders HierarchyTreeCard with level=1 ', () => {
    const { container, getByText } = render(
      <HierarchyTreeCard
        title='Shirts'
        subtitle='12 Subcategories'
        level='1'
        onClick={() => {}}
      />
    )
    fireEvent.click(getByText('Shirts'))
    expect(container).toMatchSnapshot()
  })

  it('renders HierarchyTreeCard with level=2 ', () => {
    const { container } = render(
      <HierarchyTreeCard title='Shirts' subtitle='12 Subcategories' level='2' />
    )
    expect(container).toMatchSnapshot()
  })

  it('renders HierarchyTreeCard with level=3 ', () => {
    const { container } = render(
      <HierarchyTreeCard title='Shirts' subtitle='12 Subcategories' level='3' />
    )
    expect(container).toMatchSnapshot()
  })
})
