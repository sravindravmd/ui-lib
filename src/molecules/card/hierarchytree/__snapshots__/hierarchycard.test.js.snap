// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HierarchyTreeCard/> renders HierarchyTreeCard with level=1  1`] = `
<div>
  <div
    class="sc-bdfBwQ eEhUjR"
  >
    <div
      class="sc-hKgILt jzNZGF"
      tabindex="-1"
    >
      <div
        class="sc-gsTCUz ekDpYt"
      >
        Shirts
      </div>
      <div
        class="sc-dlfnbm bMvNlL"
      >
        12 Subcategories
      </div>
    </div>
  </div>
</div>
`;

exports[`<HierarchyTreeCard/> renders HierarchyTreeCard with level=2  1`] = `
<div>
  <div
    class="sc-bdfBwQ hGCaM"
  >
    <div
      class="sc-hKgILt jzNZGF"
      tabindex="-1"
    >
      <div
        class="sc-gsTCUz cHjNND"
      >
        Shirts
      </div>
      <div
        class="sc-dlfnbm bMvNlL"
      >
        12 Subcategories
      </div>
    </div>
  </div>
</div>
`;

exports[`<HierarchyTreeCard/> renders HierarchyTreeCard with level=3  1`] = `
<div>
  <div
    class="sc-bdfBwQ hGCaM"
  >
    <div
      class="sc-hKgILt jzNZGF"
      tabindex="-1"
    >
      <div
        class="sc-gsTCUz jmnDmG"
      >
        Shirts
      </div>
      <div
        class="sc-dlfnbm bMvNlL"
      >
        12 Subcategories
      </div>
    </div>
  </div>
</div>
`;
